lib/index.ts

```diff
@@ -162,20 +162,6 @@ export async function fromImageDescriptors(
 	});
 }
 
-export function buildHasSecrets(tasks: BuildTask[]): boolean {
```

Comment:
This function was not being used/called anywhere.

Reply:
It's used in the CLI I think, or at least it should be, so we can avoid pulling the secret population images when they're not needed.

Reply:
I've restored `buildHasSecrets()` (no problem), but there are no references to it in `balena-cli`, `resin-builder` or `resin-multibuild`, neither before nor after this PR. To avoid pulling the secret population images when they're not needed, a separate test is made in `performBuilds()`, taking advantage of the fact that it already calls `generateSecretPopulationMap()`.

Reply:
Hmm I'd say just make a judgement call then @pdcastro, either remove this or change the CLI to only use this function (I think the latter is my preference but I'm easy).